tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/util/ServerUtil.java

```diff
@@ -487,23 +488,58 @@ public static PortalBundle getPortalBundle(IProject project)  throws CoreExcepti
 
         IStatus status = sdk.validate();
 
-        if ( !status.isOK())
+        if ( !status.isOK() )
         {
             return null;
         }
 
+        final String cachedAppServerPath = LiferayServerCore.getPreference(sdk.getName() + ".server.path");
+
+        final String cachedAppServerType = LiferayServerCore.getPreference(sdk.getName() + ".server.type");
+
         Map<String, Object> appServerProperties = sdk.getBuildProperties();
 
-        final PortalBundleFactory[] factories = LiferayServerCore.getPortalBundleFactories();
-        for( PortalBundleFactory factory : factories )
+        final String appServerPathString = (String) ( appServerProperties.get( "app.server.dir" ) );
```

Comment:
before we cached the appServerPath and type, we will call sdk validate method to check the build properties setting is valid or not. Only the sdk.validate() method return ok status, then we'll save all the build properties values to memory.